master/internal/db/postgres_rp_workspace_bindings.go

```diff
@@ -0,0 +1,318 @@
+	}
+	// Check if pool exists
+	poolExists := false
+	for _, pool := range resourcePools {
+		if poolName == pool.PoolName {
+			poolExists = true
+			break
+		}
+	}
+
+	if !poolExists {
+		return errors.Errorf("pool with name %v doesn't exist in config", poolName)
+	}
+
+	var bindings []RPWorkspaceBinding
+	for _, workspaceID := range workspaceIds {
+		bindings = append(bindings, RPWorkspaceBinding{
+			WorkspaceID: int(workspaceID),
+			PoolName:    poolName,
+			Valid:       true,
+		})
+	}
+
+	_, err := Bun().NewInsert().Model(&bindings).Exec(ctx)
+	return err
+}
+
+// RemoveRPWorkspaceBindings removes the bindings between workspaceIds and poolName.
+func RemoveRPWorkspaceBindings(ctx context.Context,
+	workspaceIds []int32, poolName string,
+) error {
+	if len(workspaceIds) == 0 {
+		return nil
+	}
+	// throw error if any of bindings don't exist
+	for _, workspaceID := range workspaceIds {
+		var rpWorkspaceBindings []*RPWorkspaceBinding
+		exists, err := Bun().NewSelect().Model(&rpWorkspaceBindings).Where("pool_name = ?",
+			poolName).Where("workspace_id = ?", workspaceID).Exists(ctx)
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return errors.Errorf(" workspace with id %v and pool with name  %v binding doesn't exist",
+				workspaceID, poolName)
+		}
+	}
+
+	_, err := Bun().NewDelete().Table("rp_workspace_bindings").Where("workspace_id IN (?)",
+		bun.In(workspaceIds)).Where("pool_name = ?", poolName).Exec(ctx)
+	return err
+}
+
+// ReadWorkspacesBoundToRP get the bindings between workspaceIds and the requested resource pool.
+func ReadWorkspacesBoundToRP(
+	ctx context.Context, poolName string, offset, limit int32,
+	resourcePools []config.ResourcePoolConfig,
+) ([]*RPWorkspaceBinding, *apiv1.Pagination, error) {
+	poolExists := false
+	for _, validPool := range resourcePools {
+		if validPool.PoolName == poolName {
+			poolExists = true
+			break
+		}
+	}
+	if !poolExists {
+		return nil, nil, errors.Errorf("pool with name %v doesn't exist in config", poolName)
+	}
+	var rpWorkspaceBindings []*RPWorkspaceBinding
+	query := Bun().NewSelect().Model(&rpWorkspaceBindings).Where("pool_name = ?",
+		poolName)
+
+	pagination, query, err := getPagedBunQuery(ctx, query, int(offset), int(limit))
+	if err != nil {
+		return nil, nil, err
+	}
+	// Bun bug treating limit=0 as no limit when it
+	// should be the exact opposite of no records returned.
+	// TODO: revisit and check this for commonality.
+	// We may put pagination.StartIndex-pagination.EndIndex != 0
+	// back to the function and return a nil query if StartIndex = EndIndex. This is for
+	// limit = -2, we don't run the query, return pagination only.
+	if pagination.StartIndex-pagination.EndIndex != 0 {
+		if err = query.Scan(ctx); err != nil {
+			if errors.Cause(err) == sql.ErrNoRows {
+				return rpWorkspaceBindings, pagination, nil
+			}
+			return nil, nil, err
+		}
+	}
+
+	return rpWorkspaceBindings, pagination, nil
+}
+
+// TODO find a good house for this function.
```

Comment:
What's the decision about where this should go? If it's here, we should remove the TODO.

Reply:
There hasn't been a decision yet - the original function is pretty far removed, so this is probably that should be tech debt